lib/collectors.js

```diff
@@ -2,6 +2,46 @@
 
 module.exports = (api) => {
 
+  api.metasync.collect = (expected) => {
+
+    const isCount = typeof(expected) === 'number';
+    const isKeys = Array.isArray(expected);
+    if (!(isCount || isKeys)) throw new Error('Unexpected type');
+    const keys = expected;
+    if (isKeys) expected = expected.length;
+    let count = 0;
+    let timeout = 0;
+    let timer = null;
+    let done = null;
+    const data = {};
+
+    const collector = (key, value) => {
+      if (isCount || keys.includes(key)) count++;
+      if (value instanceof Error) return done(value, data);
```

Comment:
@aqrln It seems like usage convenience of both options depends on the situation. So I propose to implement contract (key, error, value) as same as `fail` and `pick` methods